packages/table/src/table.tsx

```diff
@@ -615,6 +622,32 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         }
     }
 
+    private syncScrollPosition(_prevProps: ITableProps, prevState: ITableState) {
+        // need to be a little cute to infer how many rows and columns there were before.
+        const prevNumCols = prevState.columnWidths.length;
+        const prevNumRows = prevState.rowHeights.length;
+
+        const { numCols, numRows } = this.grid;
+        const { viewportRect } = this.state;
+
+        const maxRowIndex = numRows - 1;
+        const maxColIndex = numCols - 1;
+
+        if (numRows < prevNumRows) {
```

Comment:
This only scrolls if there are fewer rows/columns as before, but not if the rows/columns decide to shrink. Do we need to also do code for if they decide to shrink? Is that a separate commit?